packages/core/components/NumberRangePicker/index.tsx

```diff
@@ -0,0 +1,170 @@
+import { ActionButton, Spinner, SpinnerSize } from "@fluentui/react";
+import classNames from "classnames";
+import * as React from "react";
+import FileFilter from "../../entity/FileFilter";
+import { extractValuesFromRangeOperatorFilterString } from "../../entity/AnnotationFormatter/number-formatter";
+import { AnnotationValue } from "../../services/AnnotationService";
+
+import styles from "./NumberRangePicker.module.css";
+
+export interface ListItem {
+    displayValue: AnnotationValue;
+    value: AnnotationValue;
+}
+
+interface NumberRangePickerProps {
+    disabled?: boolean;
+    errorMessage?: string;
+    items: ListItem[];
+    loading?: boolean;
+    onSearch: (filterValue: string) => void;
+    onReset: () => void;
+    currentRange: FileFilter | undefined;
+    units?: string;
+}
+
+/**
+ * A NumberRangePicker is a simple form that renders input fields for the minimum and maximum values
+ * desired by the user and buttons to submit and reset the range. It also displays the
+ * overall min and max for that annotation, if available, and allows a user to select that full range.
+ *
+ * It is best suited for selecting items that are numbers.
+ */
+export default function NumberRangePicker(props: NumberRangePickerProps) {
+    const { errorMessage, items, loading, onSearch, onReset, currentRange, units } = props;
+
+    const overallMin = React.useMemo(() => {
+        return items[0]?.displayValue.toString() ?? "";
+    }, [items]);
+    const overallMax = React.useMemo(() => {
+        return items.at(-1)?.displayValue.toString() ?? "";
+    }, [items]);
+
+    const [searchMinValue, setSearchMinValue] = React.useState(
+        extractValuesFromRangeOperatorFilterString(currentRange?.value).minValue ?? overallMin
+    );
+    const [searchMaxValue, setSearchMaxValue] = React.useState(
+        extractValuesFromRangeOperatorFilterString(currentRange?.value).maxValue ?? overallMax
```

Comment:
Seems like it would be nice if FileFilter could store a `Range` object with `min` and `max` fields instead of just a string, but that's probably beyond the scope of this PR